components/raftstore/src/store/peer.rs

```diff
@@ -2335,6 +2365,8 @@ where
                 }
                 _ => {}
             }
+            // Clear the uncampaigned list.
+            self.uncampaigned_new_regions = None;
```

Comment:
Will it be cleared by becoming Candidate before becoming leader?

Reply:
It should not, which is an incorrect change after modifying the `uncampaigned_new_regions` to `Option<...>`.

Addressed.

Reply:
I'm curious why tests do not catch this.